fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/expression/ColumnExpNodeVisitor.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.database.utils.query.expression;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.ibm.fhir.database.utils.common.DataDefinitionUtil;
+import com.ibm.fhir.database.utils.query.Select;
+import com.ibm.fhir.database.utils.query.node.ExpNodeVisitor;
+
+
+/**
+ * Renders the expression node tree into a string
+ */
+public class ColumnExpNodeVisitor implements ExpNodeVisitor<Set<String>> {
+
+    @Override
+    public Set<String> paren(Set<String> expr) {
+        return expr;
+    }
+
+    /**
+     * Combine left and right sets
+     * @param left
+     * @param right
+     * @return
+     */
+    private Set<String> merge(Set<String> left, Set<String> right) {
+        Set<String> result = new HashSet<>(left);
```

Comment:
should this be a  hashset?

Reply:
No order needed. This is just trying to gather a unique set of column names used in the expression which we can then inspect to see if "TOKEN_VALUE" is part of the expression. If not, then we can apply an optimization when building the query.

Reply:
hum...  means we dedupe the columns.... might be worthy of a test case.